### PR TITLE
add boolean components

### PR DESCRIPTION
**Pull request by ctamisier**
`JhiBooleanComponent`: a angular component to display a boolean

```
<jhi-boolean [value]="job.active"></jhi-boolean>
```
screen shot 2017-11-03 at 20 06 59

----------------------------------------
```
<jhi-boolean
classFalse="fa fa-lg fa-times text-warning" 
classTrue="fa fa-lg fa-check text-primary" 
[value]="job.active">
</jhi-boolean>
```
screen shot 2017-11-08 at 22 37 35

----------------------------------------
```
<jhi-boolean 
classFalse="" 
classTrue="fa fa-lg fa-check" 
[value]="job.active">
</jhi-boolean>
```
screen shot 2017-11-08 at 22 46 31

----------------------------------------
```
<jhi-boolean
[value]="user.activated"
[textTrue]="'userManagement.activated' | translate"
textFalse="deactivated">
</jhi-boolean>
```
screen shot 2017-11-05 at 13 32 12
screen shot 2017-11-05 at 13 32 29

----------------------------------------
```
<jhi-boolean
[value]="user.activated"
classTrue="badge badge-warning"
classFalse="badge badge-info"
[textTrue]="'userManagement.activated' | translate"
textFalse="deactivated">
</jhi-boolean>
```

screen shot 2017-11-08 at 22 50 27
screen shot 2017-11-08 at 22 42 11

### CODE REVIEW

File: src/component/jhi-boolean-badge.component.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+import {Component, Input} from '@angular/core';
+
+@Component({
+    selector: 'jhi-boolean-badge',
+    template: `<span class="badge"
+               [ngClass]="{'badge-success': value, 'badge-danger': !value}"
```
Comment (deepu105) [inline src/component/jhi-boolean-badge.component.ts]: Make it possible to override the success & danger class

File: src/component/jhi-boolean.component.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+import {Component, Input} from '@angular/core';
+
+@Component({
+    selector: 'jhi-boolean',
+    template: `<i class="fa fa-lg"
+                  [ngClass]="{'fa-check': value,
```
Comment (deepu105) [inline src/component/jhi-boolean.component.ts]: Make it possible to override the icon. Look at sort directive

Comment (deepu105) [reply src/component/jhi-boolean.component.ts]: That way you can remove the nothingIfFalse option and just pass empty when particular icon is not needed

**Comment by ctamisier** [comment]
@deepu105 Thanks for your review, 
I applied your request which is indeed way better.
I did the same that for the sort directive by using the JhiConfigService and the default values.

There is one thing I didn't understand in JhiSortDirective is that we do this:
`sortIcon = 'fa-sort';` and then in the constructor we affect it again `this.sortIcon = config.sortIcon;`
So for me we lose the initial value anyway (but maybe there is something I am missing..)

I just did this my case: `classTrueConfig: string;` and no initial value... (let see if it's ok)
I'm kind of new with typescript/angular4 best practices and I have this kind of test `this.classTrue !== undefined` and that doesn't look the best for me, so a particular look on this kind of thing won't hurt.

File: src/component/jhi-boolean-badge.component.ts [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    @Input() translationKeyFalse: string;
+    clazz: string;
+    translationKey: string;
+
+    classBadgeTrueConfig: string;
+    classBadgeFalseConfig: string;
+
+    constructor(configService: JhiConfigService) {
+        const config = configService.getConfig();
+        this.classBadgeTrueConfig = config.classBadgeTrue;
```
Comment (deepu105) [inline src/component/jhi-boolean-badge.component.ts]: can you check what happens if you assign `this.classBadgeTrue = config.classBadgeTrue` here and then override it in the component usage? I dont have the time to check but I guess the input assignment happens after constructor and if so should work without using `classBadgeTrueConfig` but I could be wrong

**Comment by ctamisier** [comment]
No actually you are right! we can just assigned `this.classBadgeTrue = config.classBadgeTrue` directly, no need to use an intermediate` this.classBadgeTrueConfig` variable.
I wanted to have the same structure than the other directives and have the default values in specific fields but it's not required because it can be overriden and we can use the same `@Input` field and that's it.

So we can do the same in the other directives (JhiSortDirective, JhiSortByDirective, ...) and having for example:
`@Input() sortIcon: string;`
instead of
`sortIcon = 'fa-sort';`
offering the capability to override the default classes.

I pushed my amended commit.

File: src/component/jhi-boolean-badge.component.ts [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+        this.classBadgeTrue = config.classBadgeTrue;
+        this.classBadgeFalse = config.classBadgeFalse;
+    }
+
+    ngOnInit() {
+        this.clazz = this.value
+            ? this.classBadgeTrue
+            : this.classBadgeFalse;
+
+        this.translationKey = this.value
```
Comment (deepu105) [inline src/component/jhi-boolean-badge.component.ts]: Make sure these works when translation is disabled as well. Just check with an app without translation to be sure

**Comment by ctamisier** [comment]
Good point because it wasn't working... having this:
screen shot 2017-11-10 at 21 37 37

So I did almost the same as in jhi-item-count.component.ts.
I couldn't use `*ngIf="i18nEnabled; else noI18n"` and the `<ng-template #noI18n ...`
This was causing me an issue with `[ngClass]="clazz"` in the browser. (I still don't know why...):
```
ERROR TypeError: Cannot read property 'add' of undefined  --->JhiBooleanBadgeComponent.html:1 
    at DefaultDomRenderer2.addClass (platform-browser.es5.js?41b7:2853)
    at DebugRenderer2.addClass (core.es5.js?de3d:13679)
    at RendererAdapter.setElementClass (core.es5.js?de3d:10483)
    at eval (common.es5.js?d897:1349)
    at Array.forEach (<anonymous>)
    at NgClass._toggleClass (common.es5.js?d897:1349)
    at eval (common.es5.js?d897:1319)
    at Array.forEach (<anonymous>)
    at NgClass._applyInitialClasses (common.es5.js?d897:1319)
    at NgClass.set [as klass] (common.es5.js?d897:1230)
ERROR CONTEXT DebugContext_ {view: {…}, nodeIndex: 1, nodeDef: {…}, elDef: {…}, elView: {…}}
```

So I used `*ngIf="i18nEnabled"` and `*ngIf="!i18nEnabled"` :(

I couldn't add specific tests for this because `comp.translationKeyTrue = 'activated';` was already rendered as `<span>activated</span>`...

So now for me everything is working with an application generated with `"enableTranslation": true` or `"enableTranslation": false`.

File: src/component/jhi-boolean-badge.component.ts [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+@Component({
+    selector: 'jhi-boolean-badge',
+    template: `<span *ngIf="i18nEnabled"
+                   class="badge"
+                   [ngClass]="clazz"
+                   [jhiTranslate]="translationKey">
+               </span>
+               <span *ngIf="!i18nEnabled"
+                   class="badge"
+                   [ngClass]="clazz">{{translationKey}}
```
Comment (deepu105) [inline src/component/jhi-boolean-badge.component.ts]: I just have one more problem its weird to use translationKeyTrue/translationKeyFalse.
I can suggest a better option, leave translation out of this component and when you use the component call it using the translate pipe, something like
`<jhi-boolean-badge [value]="value" trueText"my.i18n.trueKey | translate" falseText"my.i18n.falseKey | translate" />`
This way you can decouple i18n from this component which is much better IMO

**Comment by ctamisier** [comment]
Ok for the translation, now the component is called for example with `[textTrue]="'userManagement.activated' | translate"`.

In the meantime... I also merged the 2 components so we only have `jhi-boolean`. the badge one is nothing more than a `jhi-boolean` compoment with text (`textTrue` or `textFalse`). 
And this way we only have `classTrue` and `classFalse` and the `badge` class we had can now be overriden.
I updated the first post with the examples and I pushed the changes.

**Comment by deepu105** [comment]
It looks good. Ill merge it can you add the usage examples as the component class level js doc?

**Comment by ctamisier** [comment]
sure ! https://github.com/jhipster/ng-jhipster/pull/54/files

**Pull request merged**